web/files/files.go

```diff
@@ -19,54 +22,159 @@ import (
 // DefaultContentType is used for files uploaded with no content-type
 const DefaultContentType = "application/octet-stream"
 
-// Upload is the handler for uploading a file
-//
-// swagger:route POST /files/:folder-id files uploadFile
+// DocType is the type of document, eg. file or folder
+type DocType string
+
+const (
+	// FileDocType is document type
+	FileDocType DocType = "io.cozy.files"
+	// FolderDocType is document type
+	FolderDocType = "io.cozy.folders"
+)
+
+var (
+	errDocAlreadyExists = errors.New("Directory already exists")
+	errDocTypeInvalid   = errors.New("Invalid document type")
+	errIllegalFilename  = errors.New("Invalid filename: empty or contains one of these illegal characters: / \\ : ? * \" |")
```

Comment:
By looking at Windows and its naming conventions... https://msdn.microsoft.com/en-us/library/aa365247(v=vs.85).aspx (I'm missing `>` and `<` btw)...

Don't you think it may cause us problem if one day we want to sync on a win fs ?